src/packages/emmett-mongodb/src/eventStore/mongoDBEventStore.ts

```diff
@@ -27,51 +30,100 @@ export type StreamNameParts<T extends StreamType = StreamType> = {
   streamId: string;
 };
 
-export type StreamToProject<EventType extends Event> = {
+export interface EventStreamProjection<
+  ShortInfoType extends DefaultRecord = DefaultRecord,
+> {
+  details: { name?: string };
+  short: ShortInfoType | null;
+}
+export interface EventStream<
+  EventType extends Event = Event,
+  ShortInfoType extends DefaultRecord = DefaultRecord,
+> {
+  streamName: string;
+  events: Array<ReadEvent<EventType, ReadEventMetadata>>;
+  metadata: {
+    streamId: string;
+    streamType: StreamType;
+    streamPosition: bigint;
+    createdAt: Date;
+    updatedAt: Date;
+  };
+  projections: Record<string, EventStreamProjection<ShortInfoType>>;
+}
+
+export type MongoDBProjectionHandlerContext = {
   streamName: StreamName;
-  streamType: StreamType;
-  streamId: string;
-  streamVersion: number;
-  events: ReadEvent<EventType, ReadEventMetadata>[];
+  collection: Collection<EventStream>;
 };
 
-export type Projection<EventType extends Event> = (
-  stream: StreamToProject<EventType>,
-) => void | Promise<void>;
+type MongoDBAsyncProjection<EventType extends Event = Event> = {
+  type: 'async';
+  projection: TypedProjectionDefinition<EventType>;
+};
 
-export interface EventStream<EventType extends Event = Event> {
-  streamName: string;
-  events: Array<ReadEvent<EventType, ReadEventMetadata>>;
-  createdAt: Date;
-  updatedAt: Date;
-}
-export type EventStreamEvent<EventType extends Event = Event> =
-  EventStream<EventType>['events'][number];
+type MongoDBInlineProjection<
```

Comment:
**CONCERN:** @alex-laycalvert is there a reason why you not extend the ProjectionDefinition from the regular Emmett package? See https://github.com/event-driven-io/emmett/blob/97519d0b561943a4a096f44cf95c4c8fd1393de8/src/packages/emmett/src/projections/index.ts#L30

And how PostgreSQL projection does it: https://github.com/event-driven-io/emmett/blob/97519d0b561943a4a096f44cf95c4c8fd1393de8/src/packages/emmett-postgresql/src/eventStore/projections/index.ts#L35.

Also, best if projection definition is distinct from the registration (inline, async) so someone could change whether it's run inline or async without changing the projection definition.